src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/25.diff.sql

```diff
@@ -0,0 +1,548 @@
+    SELECT * 
+	FROM sys.indexes 
+	WHERE name='IXC_ReindexJob' AND object_id = OBJECT_ID('dbo.ReindexJob'))
+BEGIN
+	DROP INDEX IXC_ReindexJob ON dbo.ReindexJob
+	WITH (ONLINE=ON)
+END
+
+EXEC dbo.LogSchemaMigrationProgress 'Adding PKC_ReindexJob'
+IF NOT EXISTS (
+    SELECT * 
+	FROM sys.key_constraints 
+	WHERE name='PKC_ReindexJob' AND type='PK')
+BEGIN
+	ALTER TABLE dbo.ReindexJob 
+	ADD CONSTRAINT PKC_ReindexJob PRIMARY KEY CLUSTERED (Id)
+	WITH (ONLINE=ON)
+END
+
+-- Deleting temporary non-clustered index created.
+EXEC dbo.LogSchemaMigrationProgress 'Dropping IX_ReindexJob'
+IF EXISTS (
+    SELECT * 
+	FROM sys.indexes 
+	WHERE name='IX_ReindexJob' AND object_id = OBJECT_ID('dbo.ReindexJob'))
+BEGIN
+	DROP INDEX IX_ReindexJob ON dbo.ReindexJob
+END
+GO
+
+-- TaskInfo
+-- Since we need to create primary key on the same of columns for which clustered index exists.
+-- We will create non-clustered index(IX) first, drop IXC, create PKC and finally drop IX
+-- so that the table would never be left without indexes.
+-- Creating temporary non-clustered index
+EXEC dbo.LogSchemaMigrationProgress 'Creating IX_TaskInfo'
+IF NOT EXISTS (
+    SELECT * 
+	FROM sys.indexes 
+	WHERE name='IX_TaskInfo' AND object_id = OBJECT_ID('dbo.TaskInfo'))
+BEGIN
+	CREATE UNIQUE NONCLUSTERED INDEX IX_TaskInfo ON dbo.TaskInfo
+	(
+		TaskId
+	)
+	WITH (ONLINE=ON)
+END
+
+EXEC dbo.LogSchemaMigrationProgress 'Dropping IXC_Task'
+IF EXISTS (
+    SELECT * 
+	FROM sys.indexes 
+	WHERE name='IXC_Task' AND object_id = OBJECT_ID('dbo.TaskInfo'))
+BEGIN
+	DROP INDEX IXC_Task ON dbo.TaskInfo
+	WITH (ONLINE=ON)
+END
+
+EXEC dbo.LogSchemaMigrationProgress 'Adding PKC_TaskInfo'
+IF NOT EXISTS (
+    SELECT * 
+	FROM sys.key_constraints 
+	WHERE name='PKC_TaskInfo' AND type='PK')
+BEGIN
+	ALTER TABLE dbo.TaskInfo 
+	ADD CONSTRAINT PKC_TaskInfo PRIMARY KEY CLUSTERED (TaskId)
+	WITH (DATA_COMPRESSION = PAGE, ONLINE=ON)
+END
+
+-- Deleting temporary non-clustered index created.
+EXEC dbo.LogSchemaMigrationProgress 'Dropping IX_TaskInfo'
+IF EXISTS (
+    SELECT * 
+	FROM sys.indexes 
+	WHERE name='IX_TaskInfo' AND object_id = OBJECT_ID('dbo.TaskInfo'))
+BEGIN
+	DROP INDEX IX_TaskInfo ON dbo.TaskInfo
+END
+GO
+
+--UriSearchParam
+-- Deleting duplicate rows based on all columns
+EXEC dbo.LogSchemaMigrationProgress 'Deleting redundant rows from dbo.UriSearchParam'
+GO
+WITH cte AS (
+    SELECT ResourceTypeId, SearchParamId, Uri, ResourceSurrogateId, IsHistory, ROW_NUMBER() 
+    OVER (
+		PARTITION BY ResourceTypeId, SearchParamId, Uri, ResourceSurrogateId, IsHistory
+		ORDER BY ResourceTypeId, SearchParamId, Uri, ResourceSurrogateId
+	) row_num
+	FROM dbo.UriSearchParam
+)
+DELETE FROM cte WHERE row_num > 1
+GO
+
+-- Create nonclustered primary key on the set of non-nullable columns that makes it unique
+EXEC dbo.LogSchemaMigrationProgress 'Adding PK_UriSearchParam'
+IF NOT EXISTS (
+    SELECT * 
+	FROM sys.key_constraints 
+	WHERE name='PK_UriSearchParam' AND type='PK')
```

Comment:
There's still clustered index "IXC_UriSearchParam" on this table? should we remove it?

Reply:
We still need this clustered index since the primary key has additional column whereas this clustered index does not.